symbol-processing-aa-embeddable/build.gradle.kts

```diff
@@ -51,6 +70,44 @@ fun String.replaceWithKsp() =
 
 val DEP_SOURCES_DIR = "$buildDir/source-jar"
 
+val validPaths = prefixesToRelocate.map {
+    it.second.split('.').filter { it.isNotEmpty() }.joinToString("/")
+} + listOf(
+    "com/google/devtools/ksp",
+    "META-INF",
+    "org/apache/log4j/package-info.class",
+    "ksp/FirNativeForwardDeclarationGetClassCallChecker.class",
+)
+
+class Trie(paths: List<String>) {
+    class TrieNode(val key: String)
+
+    private val terminals = mutableSetOf<TrieNode>()
+
+    private val m = mutableMapOf<Pair<TrieNode?, String>, TrieNode>().apply {
```

Comment:
I did not get this, if `m` is the child list of the current trie node, what is this pair used for? Feels like simply putting `String` here should be enough?

Reply:
The first element of the pair is parent. `m` is shared among all nodes and it maps (parent, key) to the child of the parent.